ldms/src/core/ldms_xprt.c

```diff
@@ -3764,20 +3764,27 @@ int ldms_xprt_sockaddr(ldms_t _x, struct sockaddr *local_sa,
 {
 	struct ldms_xprt *x = _x;
 	zap_err_t zerr;
- again:
-	if (x->sa_len) {
-		/* use cached addresses */
-		*sa_len = x->sa_len;
-		memcpy(local_sa, &x->local_sa, x->sa_len);
-		memcpy(remote_sa, &x->remote_sa, x->sa_len);
-		return 0;
-	}
-	zerr = zap_get_name(x->zap_ep, &x->local_sa, &x->remote_sa, &x->sa_len);
-	if (zerr) {
-		x->sa_len = 0;
-		return -1;
+	socklen_t __sa_len = *sa_len;
```

Comment:
local variables like this don't need the __ prefix. The symbols are not visible outside the function. I'm not sure why it's here at all actually. IF we are going to keep x->sa_len (and I'm not sure it's needed), then it the call would look something like this:
```
x->sa_len = sizeof(x->local_sa);
zerr = zap_get_name..., &x->local_sa, &x->remote_sa, &x->sa_len);
```